dynamic/dynamic.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dynamic allows the creation of dynamic (schema-less) queries.
+// The queries returned have numerous limitations and should be used for only limited use cases.
+// There are no safeguards to ensure compatibility of path and value.
+// Queries should be treated key-val pairs (not a tree structure), that is the path /a/b/c is not
+// a descendant of /a/b but rather an entirely different path. In other words, all paths are leaves
+// that contain blobs of data.
+package dynamic
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+// NewConfig creates a config query for the given path and type. The path must be gNMI path.
+// See package comment for limitations of this query type.
+func NewConfig[T any](path, origin string) (ygnmi.ConfigQuery[T], error) {
+	ps, createFn, scalar, err := newQueryField[T](path, origin)
+	if err != nil {
+		return nil, err
+	}
+
+	return ygnmi.NewLeafConfigQuery("",
+			false,
+			scalar,
+			ps,
+			createFn,
+			func() ygot.ValidatedGoStruct {
+				return nil
+			}, nil),
+		nil
+}
+
+// NewWildcard creates a wildcard query for the given path and type. The path must be gNMI path.
+// See package comment for limitations of this query type.
+func NewWildcard[T any](path, origin string) (ygnmi.WildcardQuery[T], error) {
```

Comment:
If this is supported, does this mean it's not a key-value store anymore? Why would we be expecting multiple values? How does the config side work to support multiple values being returned?

Reply:
It's still gNMI lists are supported and would work the same way. It's limited in that everything is treated as leaf.

Reply:
I see. so we're still working with a hierarchical structure then, just like the rest of YANG/gNMI?

Reply:
i took another stab at the package comment, does that explain it better?

Reply:
Yes, comments below.